microsoft-edge/devtools-guide-chromium/console/console-dom-interaction.md

```diff
@@ -5,290 +5,345 @@ author: MSEdgeTeam
 
-   ![To highlight all external links, add a green border around each](./console-dom-interaction-images/console-dom-highlight-links.png)
+   ![The Console tool, showing the list of all headings on the page, along with their URLs](./console-dom-interaction-images/heading-links.png)
 
-Instead of writing complex JavaScript to filter results, use the power of CSS selectors.
 
+#### Copy information from the Console
 
-### Creating a table
+In the previous examples, you generated lists of elements and strings of information. It can be useful to copy the generated information from the **Console** and paste it somewhere else.
 
-To create a table of the `src` and `alt` information for all images on the webpage that aren't inline images:
+To copy the return value from an expression that you run in the **Console**, use the `copy()` utility function:
 
-1. Open the **Console**.
+1. Go to the webpage that you want to extract some information from. For example, open the [DevTools Console tool DOM interactions demo page](https://microsoftedge.github.io/Demos/devtools-console-dom-interactions/) in a new tab or window.
 
-1. Paste the following code into the **Console**, and then press **Enter**:
+1. Press **Ctrl+Shift+J** (Windows, Linux) or **Command+Option+J** (macOS).  The **Console** tool opens in DevTools, next to the webpage.
+
+1. To copy the list of all image URLs on the page, enter the following code in the **Console** and press `Enter`:
 
    ```javascript
-   console.table($$('img:not([src^=data])'), ['src','alt'])
+   copy($$('img').map(img => img.src).join("\n"));
    ```
 
-   ![To programmatically select a set of elements, use a complex CSS selector](./console-dom-interaction-images/console-dom-complex-css-selector.png)
+   The `copy()` utility function can take any argument that can be converted into a string and copies it to the clipboard. In this example, the argument is a string that contains all the image URLs on the page, separated by a new line.
+
+1. Open a text editor, and then paste the copied image URLs into the editor.
+
+   ![The list of image URLs in a text editor](./console-dom-interaction-images/pasted-strings.png)
 
-<!-- mystery wiggly line preventer -->
 
+#### Clear the Console
 
-### Example: Getting all page headings and anchor URLs
+When running code in the **Console**, things can get messy.  You can clear the **Console** by using the `clear()` utility function.
 
-Ready for an even more complex example?  HTML webpages that are generated from Markdown tagging, like the present article, have automatic ID values for each heading, to allow you to deep-link directly to that section of the webpage.  For example, a `# New features` h1 heading in the Markdown source file becomes `<h1 id="new-features">New features</h1>` in the HTML file.
 
-To list all of the automatic headings to copy and paste:
+#### List the event listeners assigned to an element
```

Comment:
```suggestion
#### List the event listeners that are assigned to an element
```
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/t/that-vs-which - "Include _that_ even if the sentence is clear without it. It helps to clarify the sentence structure."